Empire - The Last Battle/Assets/Unity/Scripts/MonoBehaviours/OverworldManager.cs

```diff
@@ -307,6 +307,19 @@ void _StormshapersPlayer_OnCardRemoved(CardData card, int index)
 		}
 	}
 
+	//Won't need this function when we get rid of ModalPanel
+	void Reset()
+	{
+		StartCoroutine(SceneSwitcher.ChangeScene(Application.loadedLevelName));
```

Comment:
You should call tearDownScene before switching. otherwise the event listeners won't be removed and you will have errors in the newly loaded scene due to multiple events being fired.